samza-api/src/main/java/org/apache/samza/operators/functions/FilterFunction.java

```diff
@@ -22,7 +22,8 @@
 
 
 /**
- * A function that specifies whether a message should be retained for further processing or filtered out.
+ * Specifies whether a message should be retained for further processing.
```

Comment:
No, cannot be used interchangeably.
We discussed using java types for these functions (Predicate, Function, BiFunction etc.) instead of introducing our own earlier (see https://github.com/apache/samza/pull/51/files/001be632dcfdbbb138b792265b991722a2ea5597#diff-e1f7020353d9bd97ac274b6e3607d5e3 for more context). TL;DR currently these give us some flexibility to add marker interfaces or behavior (Initable, Closeable, Serializable) to these functions if we need it. For users that just use a lambda the difference shouldn't be noticeable.
I think we have a clearer expectations from these functions now, so we can discuss this again when @nickpan47's back.